### PR TITLE
VDAF-07 and DAP-07

### LINKED ISSUE #100: Wire up `Prio3Aes128CountVec`
Divvi Up should allow tasks to use [`prio::vdaf::prio3::Prio3Aes128CountVec`](https://docs.rs/prio/0.8.4/prio/vdaf/prio3/type.Prio3Aes128CountVec.html).

### LINKED ISSUE #359: DAP-07/VDAF-07 compatibility
The new drafts primary deal with the aggregation sub-protocol, but there are some relevant changes for the client:

- [x] Domain separation strings
- [x] `struct Report` now has exactly two encrypted input shares
- [x] SHAKE128
- [x] Prio3SumVec
- [x] Prio3Histogram

[DAP-04 to DAP-07 diff](https://author-tools.ietf.org/iddiff?url1=draft-ietf-ppm-dap-04&url2=draft-ietf-ppm-dap-07&difftype=--html)
[VDAF-05 to VDAF-07 diff](https://author-tools.ietf.org/iddiff?url1=draft-irtf-cfrg-vdaf-05&url2=draft-irtf-cfrg-vdaf-07&difftype=--html)

### PR DESCRIPTION
**Pull request by jbr**
closes #359
closes #100

### CODE REVIEW

File: packages/interop-test-client/src/index.spec.ts [hunk 1]
```diff
@@ -6,9 +6,10 @@ import { spawnSync } from "node:child_process";
 import { Server } from "node:http";
 
 const JANUS_INTEROP_AGGREGATOR_IMAGE =
-  "us-west2-docker.pkg.dev/divviup-artifacts-public/janus/janus_interop_aggregator:0.5.2@sha256:15394dcc4d6eacf4e2112ef0d15a972328a7d805cb39ba9fc607b6a3e62fa9df";
+  "us-west2-docker.pkg.dev/divviup-artifacts-public/janus/janus_interop_aggregator@sha256:8cc873f7a8be459fe2dbecdf78561806b514ac98b4d644dc9a7f6bb25bb9df02";
```
Comment (divergentdave) [inline packages/interop-test-client/src/index.spec.ts:9]:
```suggestion
  "us-west2-docker.pkg.dev/divviup-artifacts-public/janus/janus_interop_aggregator:0.6.3@sha256:8cc873f7a8be459fe2dbecdf78561806b514ac98b4d644dc9a7f6bb25bb9df02";
```

Comment (divergentdave) [reply packages/interop-test-client/src/index.spec.ts]: (sorry, I fat-fingered "add single comment" when I meant to cancel a comment) Could you do the same on the image below?

File: packages/vdaf/src/index.ts [hunk 2]
```diff
@@ -98,12 +115,51 @@ export abstract class Vdaf<
...
+
+  encodePublicShare(publicShare: PublicShare): Buffer {
+    publicShare;
+    return Buffer.alloc(0);
+  }
+
+  encodeAggregatorShare(aggregatorShare: AggregatorShare): Buffer {
+    aggregatorShare;
+    return Buffer.alloc(0);
+  }
```
Comment (divergentdave) [inline packages/vdaf/src/index.ts:131]: Should these be abstract instead? (same for the test vector methods returning empty strings or buffers below)

**Review by divergentdave — APPROVED** [review]

File: packages/vdaf/src/index.ts (hunk 2 above)
Comment (jbr) [reply packages/vdaf/src/index.ts:131]: I wasn't sure if these should be required, which they would be if they were abstract. This was related to my confusion about serialization, vdaf, prio3, and dap.

Comment (jbr) [reply packages/vdaf/src/index.ts]: made them abstract (and required) and added a few other encoding signatures to vdaf

**Pull request merged**